src/basic/log.c

```diff
@@ -250,6 +250,8 @@ static bool stderr_is_journal(void) {
 int log_open(void) {
         int r;
 
+        PROTECT_ERRNO;
```

Comment:
maybe add a brief comment here, to explain this, since this might be surprising:

```c
/* Let's make sure we leave errno unmodified here. This call is often called in similar contexts as the usual log
   functions as preparation for being able to log, and hence should not modify errno either so that an immediately
   following log message can still reference an error easily that happened immediately before this call. */
```

or something like that

Reply:
Added a slightly different (& hopefully slightly clearer) comment :slightly_smiling_face: